prime-router/src/main/kotlin/azure/SettingsFunctions.kt

```diff
@@ -382,4 +379,21 @@ open class BaseFunction(
     }
 
     private fun errorJson(message: String): String = HttpUtilities.errorJson(message)
+
+    /**
+     * Run authentication then filters out non-admin claims
+     * @param request Http request to authenticate
+     * @return Authenticated claims if primeadmin is authorized, else null
+     */
+    private fun authenticateAdmin(
```

Comment:
Optional nit:  this might be a good function to move onto the authenticated claims class